Add reference TSV file with cell type annotations and ontology IDs for SCimilarity labels

allyhawkins opened the pull request:
### Purpose/implementation Section
#### Please link to the GitHub issue that this pull request addresses.

Closes #1254 

#### What is the goal of this pull request?

The goal of this PR is to create a reference TSV file with the cell type annotations and ontology IDs found in the `SCimilarity` foundational model. We can then use this to create an udpated consensus cell type reference. 

#### Briefly describe the general approach you took to achieve this goal.

- The first thing I did was download the zip file with the `SCimilarity` model from [`zenodo`](https://zenodo.org/records/10685499). I downloaded the most recent version since the file names match what's outlined in the [tutorial for cell type annotation](https://genentech.github.io/scimilarity/notebooks/cell_annotation_tutorial.html). As a side note, I did download the older smaller file for annotation and the labels ended up being equivalent, but I think we should still use the recent version. 
- I was originally going to follow the tutorial to read in the model with `SCimilarity` and then grab the annotations to export, but when I was digging around the files, I found the `annotation/reference_labels.tsv` that matched the annotations listed in the tutorial. So I decided to just write an R script that reads in that file, grabs the associated cell ontology ID for each annotation and then exports it as a TSV file. This TSV file won't actually be used to run the analysis, but I plan to use it to assign ontology IDs to the final annotations and for assigning consensus cell types. 
- I initiated the `run-analysis.sh` script here to include downloading and uncompressing the model (if it doesn't already exist) and then creating the reference TSV file. 

A few caveats: 

- There are 9 labels that did not have a direct match in cell ontology. I created a separate TSV file for those 9 labels and found the most appropriate human readable value that matches. For all but `native cell` and `animal cell` there was a direct synonym. For `native cell`, I just used `cell` and then `animal cell` I used `eukaryotic cell`. 
- There is a suggestion in the annotation tutorial about subsetting the reference to just cell types we want. Jackie and I had talked about not doing that, but I don't love having `cell` or `eukaryotic cell` in the reference? I think we should still include it here since its just a reference table but when we actually run this we might think about subsetting. Here's the text from the tutorial about this: 
> Alternatively, we can subset the reference to just the cell types we want to classify to. This also reduces noise in cell type annotation.

> Note that subsetting can slow classification speeds as the kNN is optimized for the full reference.


#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes

### Provide directions for reviewers

#### What are the software and computational requirements needed to be able to run the code in this PR?

I updated renv here to include the packages needed for mapping ontology IDs. 
Also, the model file is ~ 30 GB so you need to have enough space if you want to download it. 



### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._


#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [x] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

sjspielman commented on analyses/cell-type-scimilarity/.Rprofile:
```diff
@@ -1,3 +1,4 @@
+source("renv/activate.R")
```
renv got aggressive here i think?

sjspielman commented on analyses/cell-type-scimilarity/README.md:
```diff
@@ -4,9 +4,15 @@
 
 This module will be used to annotate cell types in all ScPCA libraries using [`SCimilarity`](https://genentech.github.io/scimilarity/index.html). 
 
+For cell type annotation, we use the model available on [Zenodo](https://zenodo.org/records/10685499). 
```
Just want to note for #1266, from Zenodo

> This release of the model is intended for use with v0.2.0 of the SCimilarity API.

sjspielman reviewed (commented):
Just was skimming through, left tiny comments for fun!

jashapiro commented on analyses/cell-type-scimilarity/scripts/01-assign-ontology-ids.R:
```diff
@@ -0,0 +1,86 @@
+  dplyr::left_join(ontology_labels_df, by = "human_readable_value") |> 
+  dplyr::rename(scimilarity_celltype_ontology = ontology_id) 
+
+# check that all labels have an ontology id 
+stopifnot(
+  "Ontology IDs cannot be found for all cell type annotations" = 
+    sum(is.na(labels_df$scimilarity_celltype_ontology)) == 0 
+)
```
Instead of just quitting here, I would do one of two things (or both).
1. Write out a list of the missing terms.
2. Still create the output file with the NA values.

allyhawkins commented:
@jashapiro this should be ready for another look! 

- I updated the shell script to be `setup-analysis.sh` and updated documentation based on that change. I'll file a separate issue to track adding this to CI and note the current limitations there. 
- I modified the script for generating the TSV file to read in the annotations file directly and not just the model directory. 
- The script also now has a check for any missing annotations and prints them out rather than quitting. Either way I do write the file out.

jashapiro reviewed (approved):
LGTM!

sjspielman commented on analyses/cell-type-scimilarity/scripts/01-assign-ontology-ids.R:
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env Rscript
+
+# This script is used to match cell types present in the SCimilarity model to cell ontology IDs
+# The output is a TSV file that contains `scimilarity_celltype_ontology`, `scimilarity_celltype_annotation`, and `human_readable_value`
```
Might want to indicate that this exports 2 tsv files

allyhawkins replied:
It doesn't export 2 TSV files, it only exports the one TSV file with the assigned ontology IDs. The other file I made manually and saved to the repo.

sjspielman replied:
There are definitely two tsvs in the `make_options` which is why I commented (full disclosure I didn't scroll past there 😬), so I guess the `missing` one is stale code and should be removed from the script

sjspielman commented on analyses/cell-type-scimilarity/references/README.md:
```diff
@@ -0,0 +1,15 @@
+
+This folder contains any reference files used for cell type annotation. 
+
+* `scimilarity-missing-ontologies.tsv`: This file contains a table of all annotations that are part of the [`model_v1.1`](https://zenodo.org/records/10685499) that do not have a direct cell ontology match. 
```
can we also say that the same script exported this tsv too? (as in the next bullet?)
